### PR TITLE
BZ-1769667: Simultaneous drain of two OCS nodes

### PR DESCRIPTION
**Pull request by PrasadDesala**
This patch automates BZ-1769667: Simultaneous drain of two OCS nodes (from different AZs) makes one drain operation stuck for infinite time.

Signed-off-by: Prasad Desala <tdesala@redhat.com>

### CODE REVIEW

File: tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+)
+from tests.helpers import wait_for_resource_state, label_worker_node
+from ocs_ci.ocs.resources import pod
+
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(autouse=True)
+def Delete_unscheduled_nodes_teardown(request):
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py]: change to lower case - delete_unscheduled_nodes_teardown

File: tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+import logging
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py]: I suggest we have all tests related to node drain to be under the same test file (test_node_maintenance.py)

File: tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py (hunk 1 above)
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py]: the implementation already exists in tests.manage.z_cluster.nodes.test_nodes_maintenance.schedule_nodes_teardown

Having this test in this file will prevent this duplicity

File: tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            ) == constants.NODE_READY_SCHEDULING_DISABLED
+        ]
+        remove_nodes(unscheduled_nodes)
+    request.addfinalizer(finalizer)
+
+
+@tier4
+@tier4c
+@ignore_leftovers
+@aws_platform_required
```
Comment (vasukulkarni) [inline tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py]: Is aws limitation still required? same for ipi as well?

File: tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py (hunk 1 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py]: Moved this file to test_nodes_maintenance.py

File: tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py (hunk 2 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py]: Done

File: tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py (hunk 1 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py]: Done

File: tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py (hunk 3 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_simultaneous_drain_of_two_ocs_nodes.py]: In this test we are adding nodes by increasing machine set. So, this is specific to aws for now.
I will add UPI, once the functions for add_node are ready.

**Comment by ebenahar** [comment]
@PrasadDesala please rebase

**Comment by PrasadDesala** [comment]
> @PrasadDesala please rebase

Done.

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py [hunk 4]
```diff
@@ -3,38 +3,57 @@
...
 )
 from ocs_ci.framework.testlib import (
     tier1, tier2, tier3, tier4, tier4b,
     ManageTest, aws_platform_required, ignore_leftovers, bugzilla
 )
 
 from tests.sanity_helpers import Sanity
+from ocs_ci.ocs.resources import pod
+from tests.helpers import (
+    label_worker_node, remove_label_from_worker_node)
```
Comment (RazTamir) [inline tests/manage/z_cluster/nodes/test_nodes_maintenance.py:25]: ) in new line

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py [hunk 5]
```diff
@@ -3,38 +3,57 @@
...
+def teardown(request):
     """
-    Make sure that all cluster's nodes are in 'Ready' state and if not,
-    change them back to 'Ready' state by marking them as scheduble
+
+    Tear down function
 
     """
     def finalizer():
+        # Make sure that all cluster's nodes are in 'Ready' state and if not,
```
Comment (RazTamir) [inline tests/manage/z_cluster/nodes/test_nodes_maintenance.py:40]: function docstring should be in:
```
"""
Make sure that all cluster's nodes are in 'Ready' state and if not,
change them back to 'Ready' state by marking them as schedulable
"""
```

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py [hunk 6]
```diff
@@ -222,3 +241,126 @@ def test_2_nodes_different_types(self, pvc_factory, pod_factory):
...
+                marks=pytest.mark.polarion_id("OCS-2129")
+            ),
+        ]
+    )
+    def test_simultaneous_drain_of_two_ocs_nodes(
+        self, pvc_factory, pod_factory, dc_pod_factory,
+        interface
+    ):
+        """
+
```
Comment (RazTamir) [inline tests/manage/z_cluster/nodes/test_nodes_maintenance.py:266]: needed?

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py [hunk 7]
```diff
@@ -222,3 +241,126 @@ def test_2_nodes_different_types(self, pvc_factory, pod_factory):
...
+            )
+            pod.run_io_in_bg(dc_pod, fedora_dc=True)
+            dc_pod_obj.append(dc_pod)
+
+        # Get the machine name using the node name
+        machine_names = [
+            machine.get_machine_from_node_name(osd_running_worker_node)
+            for osd_running_worker_node in osd_running_worker_nodes[:2]
+        ]
+        log.info(f"{osd_running_worker_nodes} associated "
```
Comment (RazTamir) [inline tests/manage/z_cluster/nodes/test_nodes_maintenance.py:299]: please fix alignment

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py (hunk 4 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_nodes_maintenance.py]: Done

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py (hunk 5 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_nodes_maintenance.py]: Done

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py (hunk 6 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_nodes_maintenance.py]: Added test case steps.

File: tests/manage/z_cluster/nodes/test_nodes_maintenance.py (hunk 7 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_nodes_maintenance.py]: Done

**Pull request merged**